parser/src/parser/mod.rs

```diff
@@ -1582,6 +1583,45 @@ impl<'src> ParserImpl<'src> {
             .end()
     }
 
+    /// Parses `with` expression.
+    ///
+    /// ```text
+    /// WITH_EXPR :=
+    ///     `with` WITH_IDENTIFIERS `:`
+    ///         `(`
+    ///             BOOLEAN_EXPR
+    ///         `)`
+    /// ```
+    fn with_expr(&mut self) -> &mut Self {
+        self.begin(WITH_EXPR)
+            .expect(t!(WITH_KW))
+            .then(|p| p.with_identifier())
+            .expect(t!(COLON))
+            .expect(t!(L_PAREN))
+            .then(|p| p.boolean_expr())
+            .expect(t!(R_PAREN))
+            .end()
+    }
+
+    /// Parses `with` identifiers.
+    ///
+    /// ```text
+    /// WITH_IDENTIFIERS :=
+    ///     IDENT `=` EXPR (`,` IDENT `=` EXPR)*
+    ///
+    fn with_identifier(&mut self) -> &mut Self {
+        self.begin(WITH_IDENTIFIERS)
```

Comment:
I think `WITH_IDENTIFIERS` is not the best name for this. Something like `WITH_DECLARATIONS` is more appropiate, or `WITH_DECLS` for coherency with other names.

Reply:
I have unified it across the code under `declarations`, it should be all good now